Automattic/wp-calypso
# Marketplace: Add category result text / title and clear link

#### Changes proposed in this Pull Request

* Adds "Found X plugins. Clear" text for categories similar to the text added for search results in https://github.com/Automattic/wp-calypso/pull/63645
* Cleans up css
* Removes excessive margins between search and category selector cc @SaxonF fyi in case this wasn't bug

#### Testing instructions

* http://calypso.localhost:3000/plugins
* Selecting a category should display the result count and a clear button
* Clear link should clear the selected category
* Note, on mobile clicking clear won't scroll category to the start to show discover selected again, this is fixed in https://github.com/Automattic/wp-calypso/pull/63842 (likely gets merged first anyway)

Before
![Screenshot 2022-05-20 at 15-23-05 Plugins ‹ Site Title — WordPress com](https://user-images.githubusercontent.com/811776/169456309-33955ede-ece8-41ce-a2a9-7e6d64c0e5b4.png)

After
![Screenshot 2022-05-23 at 15-46-53 Plugins — WordPress com](https://user-images.githubusercontent.com/811776/169751672-c2af7122-2f5d-4509-b1fc-19489cdd57c4.png)


Fixes https://github.com/Automattic/wp-calypso/issues/63839 
Fixes https://github.com/Automattic/wp-calypso/issues/63791
